e2e/client/ippool.go

```diff
@@ -14,7 +14,7 @@ import (
 func isIPPoolAllocationsEmpty(k8sIPAM *kubeClient.KubernetesIPAM, ipPoolName string) wait.ConditionFunc {
 	return func() (bool, error) {
 		ipPool, err := k8sIPAM.GetIPPool(context.Background(), ipPoolName)
-		if err.Error() == kubeClient.ErrMsgK8sPoolInit {
+		if err != nil && err.Error() == kubeClient.ErrMsgK8sPoolInit {
```

Comment:
this wouldn't be needed if you just used the errors API - i.e. `errors.As` .

Reply:
> this wouldn't be needed if you just used the errors API - i.e. `errors.As` .

I corrected, only what fell with errors and did not pass tests without optimizations and logic replacement
```
Run go vet --tags=test ./...
  go vet --tags=test ./...
  shell: /usr/bin/bash -e {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.19.1/x64
# github.com/k8snetworkplumbingwg/whereabouts/e[2](https://github.com/kmlebedev/whereabouts/actions/runs/3184060412/jobs/5192029033#step:9:2)e/client
Error: e2e/client/ippool.go:20:[6](https://github.com/kmlebedev/whereabouts/actions/runs/3184060412/jobs/5192029033#step:9:7): second argument to errors.As should not be *error
Error: Process completed with exit code 2.
```

https://github.com/kmlebedev/whereabouts/actions/runs/3184060412/jobs/5192029033

Reply:
Yes, but that happens because we're currently passing a pointer to an error instead of the error itself. 

I am nearly sure if you do the following `go vet` will **not** complain:
```golang
		noPoolError := fmt.Errorf("k8s pool initialized")
		if errors.As(err, noPoolError) {
		...
```
Above I just removed the `&` from `noPoolError`.